pkg/services/folder/folderimpl/folder.go

```diff
@@ -106,20 +108,33 @@ func ProvideService(
 	ac.RegisterScopeAttributeResolver(dashboards.NewFolderUIDScopeResolver(srv))
 
 	if features.IsEnabledGlobally(featuremgmt.FlagKubernetesFoldersServiceV2) {
-		k8sHandler := &foldk8sHandler{
-			gvr:                    v0alpha1.FolderResourceInfo.GroupVersionResource(),
-			namespacer:             request.GetNamespaceMapper(cfg),
-			cfg:                    cfg,
-			restConfigProvider:     apiserver.GetRestConfig,
-			recourceClientProvider: unified.GetResourceClient,
```

Comment:
I wonder if we can get rid of `unified.GetResourceClient` now. I don't think there will be any other packages using this method after merging this PR.

I can work on a follow-up PR for this, though

Reply:
The k8s client now is using it (for dashboards as well), since we can't have the resource client passed into the folder service to pass into the k8s client

Reply:
Ah!!!! Makes sense!